src/aleph/http/http2.clj

```diff
@@ -49,102 +91,356 @@
 (def ^:const max-allowed-frame-size 16777215) ; 2^24 - 1, see RFC 9113, SETTINGS_MAX_FRAME_SIZE
 (def ^:dynamic *default-chunk-size* 16384) ; same as default SETTINGS_MAX_FRAME_SIZE
 
+;; TODO: optimize. Either non-concurrent HashMap, regex, or with a fn
```

Comment:
What needs to be optimized?

Reply:
`invalid-headers` is checked on every single header, so it's a hot spot. Should probably replace with a HashSet. It's ~3x faster, and we don't need to synchronize, since it will never be updated after initialization.

Reply:
Alright, I turned it into a HashSet

Reply:
TIL: I always considered Clojure Sets being HashSet under the hood.

Reply:
All non-sorted structures are variants of Bagwell's Hash Array Mapped Tries.

Sorted maps/sets in clj are persistent versions of Red-Black trees with 2-way/binary branching from Okasaki's _Purely Functional Data Structures_.

HashSets aren't "persistent", which means to make them immutable, you'd have to do full copies of all data, which makes immutability very expensive (even more than it is with persistent data structs)